### ISSUE TITLE
Abandoned?

### ISSUE THREAD

**Issue by csvan** [SheetJS/js-xlsx#513]
I hate to ask this, but is there any hope at all for PRs being accepted and further work being done on this project? Or should we just move ahead with a community fork?

**Comment by trendoid** [comment]
I just forked it also.  I'm number #874.  When there are that may PRs, I just give up on anyone answering.

**Comment by eddyparkinson** [comment]
@varblob @elad @lostinplace @altkatz 
>I hate to ask this, but is there any hope at all for PRs being accepted and further work being done on this
>project? Or should we just move ahead with a community fork?

**Comment by varblob** [comment]
I'm not a core dev, just submitted a typo fix but it looks pretty dead to me...

**Comment by tarwich** [comment]
Sorry I missed all this. I've forked the repo and I'm trying to maintain PRs. I've asked the author in #497 to allow us to help by either adding us to the repo, or blessing mine. In the meantime, you can PR at https://github.com/tarwich/js-xlsx and I will merge them.

**Comment by eddyparkinson** [comment]
email from sheetjs:

> tl;dr: you should fork and make your own modules.  
> 
> We still use the modules as they stand every day in the context of a few server processes as well as a browser tool based on http://oss.sheetjs.com/js-xlsx/.  Other people apparently use it too: xlsx, for example, shows 3599 downloads yesterday 21703 over the last week and 81971 over the last month according to https://www.npmjs.com/package/xlsx.
> 
> Making a change is not an easy task, especially on the write side, because the files have to work with many versions of excel as well as with third-party tools like LibreOffice which don’t even bother with full Excel compatibility.   For example, https://bugs.documentfoundation.org/show_bug.cgi?id=83511 came up as a result of a user report and it effectively boiled down to LibreOffice conveniently deciding that it is appropriate to chop of the last few bits of every result.  A math program decided that it was appropriate to fudge the numbers :(  So you have a series of workarounds to address them and a series of seemingly nonsensical hacks to work around undocumented limitations such as Excel XML parsing logic.
> 
> To be absolutely clear about testing, we check some stuff manually against up to 13 versions of Excel, depending on the feature.  For example, here is Excel 2.0: http://i.imgur.com/cy05pPz.png.  It is an incredibly tedious and thankless task to slog through the relevant versions of Excel and test manually, but that’s what it takes.  Third party tools generate files based on different versions of Excel, so a tool that reads files should be able to handle every format and a tool that writes files should try to generate files that are valid with as many versions as appropriate.
> 
> There is a myriad of ways to make small changes that work for some versions of excel but not others, or work for some versions of excel but not with other commonly-used tools.  Even under the format of XLSX there are four main versions: 2007 XLSX, 2010 XLSX, 2013 XLSX, 2016 XLSX.  New features that work in 2016 XLSX may not work as expected in older versions.   So there’s a tradeoff, and while some developers throw caution to the wind we have decided to be a lot more conservative in changes.
> 
> If you would prefer breaking support for certain versions or formats, you are free to fork the open source modules and publish to NPM under different names.

**Comment by tarwich** [comment]
Wow. That's a great response. Glad they took the time to explain all that.

The biggest thing I'm hearing is that better testing is needed – and I agree. In the meantime, I'll still take any PR that comes my way. And... I'll update my request to acknowledge this information.

**Comment by eddyparkinson** [comment]
I am mostly interested in exporting excel format files, as ethercalc gets requests to export formulas. Exporting style would also be good. I am guessing the reading side is of high quality, as I never see anyone saying it fails to read.

**Comment by elad** [comment]
@tarwich @eddyparkinson sorry for the delay, I hadn't noticed you pinged me directly.

I wrote some of the style handling code and possibly some bits that have to do with date handling. However, I'm not an official maintainer so I can't comment on status and don't know anything you guys don't. I tried pinging the author(s) but have had very little success.

I think this initiative (of forking) is fantastic, but please try doing so together with the existing fork from @protobi (https://github.com/protobi/js-xlsx). Pieter has done a lot of work on this codebase and, to the best of my knowledge, actively maintains his fork. No need in doubling community efforts. :)

**Comment by TheBrianJ** [comment]
@eddyparkinson sounds like the maintainers need time.  If we put together a small bounty, we might be able to buy some of their time to work through the PRs.

**Comment by eddyparkinson** [comment]
**Organization account**
@elad tarwich created an Organization account - https://github.com/node-xlsx
I invited pietersv -let me know if you want adding.

https://github.com/blog/674-introducing-organizations

**Comment by SheetJSDev** [comment]
Hello All!

Now that [support for 0.12 is ending](https://github.com/nodejs/LTS#lts-schedule) and we've moved up to 6.0, we are starting to roll out fixes to support both older and newer versions of node.  The just-published `0.8.1` bumps the codepage version, and that should address the majority of node version issues.  There are some rough edges but those will require fixes in the dependency tree.

We are truly blown away by the general popularity and clever use cases for this and the related projects.  It has grown far beyond our wildest expectations!  To push the project to the next level, we will have to systematically re-evaluate every component.  For example, given the string formatting logic employed by Excel, the [actual code](https://github.com/SheetJS/ssf/blob/master/ssf.js) is unsurprisingly convoluted.  We have some ideas for improvement based on our work [implementing POSIX printf in JS](https://github.com/SheetJS/printj), but it takes a significant amount of our volunteered time and resources to work through the details.

Pull requests and code contributions are welcome, but please understand that while the github UI makes merging look simple, the interface masks the social complexity.  To put it simply, once any code becomes part of the codebase, the maintenance responsibility shifts to us.  We cannot blame an external contributor for an issue.  The process of making sure that changes don't cause issues starts from understanding and stressing existing tools to grasp the scope of proposed changes and culminates in a series of test cases to ensure that we have, to the best of our knowledge and experience, considered all possible circumstances.  

One other issue for consideration, that we did not mention in the email quoted in this thread, is IP and licensing.  https://github.com/SheetJS/js-xls/issues/1#issuecomment-13852286 is a summary of the relevant issues from our perspective. https://github.com/kennethreitz/tablib/issues/114 includes a more thorough analysis of the open source landscape.  Basically, another JS library for XLSX files [has a platform limitation clause](https://github.com/stephen-hardy/xlsx.js/blob/master/LICENSE.txt#L22):

>  (F) Platform Limitation- The licenses granted in sections 2(A) & 2(B) extend only to the software or derivative works that (1) are run on a Microsoft Windows operating system product, and (2) are not Excluded Products.

All of this is to say that we must be vigilant in ensuring that all contributions are clean.  The source code has many references to parts of the various specifications so we can justify most of the decisions on the basis of specifications or test cases.

Should you find a specific need for a change that has not been made to the main code repository or npm module, feel free to follow the established github and npm practices of forking modules and publishing to your own name space.   Attempting to take down a github organization or wrest control of npm modules ultimately discourages future efforts.
